backend/src/user/user.controller.ts

```diff
@@ -50,4 +55,19 @@ export class UserController {
     }
     await this.userService.updateUserDetails(userId, data);
   }
+
+  @UseGuards(JwtAuthGuard)
+  @Get(":userId(\\d+)")
+  async getUserName(
+    @Param("userId", ParseIntPipe) userId: number,
+  ): Promise<GetUserNameRes | null> {
+    const User = await this.userService.getById(userId);
+    if (User?.name != null) {
+      const responseBody: GetUserNameRes = {
+        name: User?.name,
+      };
+      return responseBody;
+    }
+    return null;
```

Comment:
Guard clauses are a little more clean and we leave PascalCase for class or interface names.

(Not 100% sure if the shorthand in the last line works, otherwise might need `{ name: name }`)

```suggestion
    const user = await this.userService.getById(userId);
    if (!user) {
     return null;
    }
    const { name } = user;
    return { name };
```